crypto/src/proofs/transparent.rs

```diff
@@ -56,63 +57,46 @@ impl SpendProof {
             return Ok(());
         }
 
-        // Note commitment integrity.
-        let s_component_transmission_key = self.note.transmission_key_s();
-        let note_commitment_test = note::commitment(
+        gadgets::note_commitment_integrity(
             self.note.note_blinding(),
             self.note.value(),
             self.note.diversified_generator(),
-            s_component_transmission_key,
-            self.note.clue_key(),
-        );
-
-        if self.note_commitment_proof.commitment() != note_commitment_test {
-            return Err(anyhow!("note commitment mismatch"));
-        }
+            self.note.transmission_key_s(),
+            self.note.clue_key().clone(),
```

Comment:
Is there a reason for the `note_commitment_integrity` gadget not to take the `note` itself, rather than all of its components?